agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -140,6 +179,32 @@ def ensure_started
       raise exc
     end
 
+    def ensure_healing
+      if health.count < 5
+        info "restarting #{service_pod.name} because it's unhealthy"
+        restart_instance
+      elsif health.count == 5
+        info "re-creating #{service_pod.name} because it's unhealthy"
+        ensure_running(pull: false)
+      elsif health.count % 5 == 0
```

Comment:
Yeah, I left this braindead logic here on purpose. Should agent backoff like in restarts or something else?